### PR TITLE
DI: Now uses Dependency Injection

### PR DESCRIPTION
**Pull request by BriWak**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala [hunk 1]
```diff
@@ -36,38 +37,29 @@ import scala.concurrent.ExecutionContext.Implicits.global
...
-  override val environment: String = getConfString("des-hod.environment", "")
+class DesConnector @Inject()(environment: Environment,
+                             configuration: Configuration,
+                             desSummaryRepository: DesSummaryRepository,
+                             desNIRecordRepository: DesNIRecordRepository,
+                             desLiabilitiesRepository: DesLiabilitiesRepository,
+                             metrics: MetricsService) extends ServicesConfig {
 
+  val serviceUrl: String = baseUrl("des-hod")
+  val authToken: String = s"Bearer ${getConfString("des-hod.authorizationToken", "")}"
```
Comment (matthew-mcnally) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala]: You could get this from the Configuration

File: app/uk/gov/hmrc/nationalinsurancerecord/controllers/DocumentationController.scala [hunk 2]
```diff
@@ -16,13 +16,15 @@
...
+import com.google.inject.{Inject, Singleton}
+import play.api.http.HttpErrorHandler
 import play.api.mvc.{Action, AnyContent}
 import uk.gov.hmrc.nationalinsurancerecord.config.{APIAccessConfig, AppContext}
 import uk.gov.hmrc.nationalinsurancerecord.domain.APIAccess
 import uk.gov.hmrc.nationalinsurancerecord.views._
 
-class DocumentationController (errorHandler: HttpErrorHandler, appContext: AppContext)
+@Singleton
+class DocumentationController @Inject()(errorHandler: HttpErrorHandler, appContext: AppContext)
```
Comment (matthew-mcnally) [inline app/uk/gov/hmrc/nationalinsurancerecord/controllers/DocumentationController.scala]: Needs module to bind LazyHttpErrorHandler

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala (hunk 1 above)
Comment (BriWak) [reply app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala]: Done

File: app/uk/gov/hmrc/nationalinsurancerecord/controllers/DocumentationController.scala (hunk 2 above)
Comment (BriWak) [reply app/uk/gov/hmrc/nationalinsurancerecord/controllers/DocumentationController.scala]: As discussed this is already bound elsewhere

**Pull request merged**